client/store/reports/stats/test/reducer.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * @format
+ */
+
+/**
+ * External dependencies
+ */
+import deepFreeze from 'deep-freeze';
+
+/**
+ * Internal dependencies
+ */
+import { ERROR } from 'store/constants';
+import reportStatsReducer from '../reducer';
+import { getQueryKey } from 'store/util';
+
+describe( 'reportStatsReducer()', () => {
+	it( 'returns an empty data object by default', () => {
+		const state = reportStatsReducer( undefined, {} );
+		expect( state ).toEqual( {} );
+	} );
+
+	it( 'returns with received report data', () => {
+		const originalState = deepFreeze( {} );
+		const query = {
+			after: '2018-01-04T00:00:00+00:00',
+			before: '2018-07-14T00:00:00+00:00',
+			interval: 'day',
+		};
+		const report = {
+			totals: {
+				orders_count: 10,
+				num_items_sold: 9,
+			},
+			interval: [ 0, 1, 2 ],
+		};
+		const endpoint = 'revenue';
+
+		const state = reportStatsReducer( originalState, {
+			type: 'SET_REPORT_STATS',
+			endpoint,
+			query,
+			report,
+		} );
+
+		const queryKey = getQueryKey( query );
+		expect( state[ endpoint ][ queryKey ] ).toEqual( report );
+	} );
+
+	it( 'tracks multiple queries per enpdoint in report data', () => {
```

Comment:
We should throw in a test that uses two different endpoints for good measure.

Also typo here. `enpdoint` -> `endpoint`